sympy/printing/fcode.py

```diff
@@ -129,6 +131,31 @@ def _lead(self):
         else:
             raise ValueError("Unknown source format: %s" % self._settings['source_format'])
 
+    def _print_Symbol(self, expr):
+        if self._settings['name_mangling'] == True:
+            if expr not in self.mangled_symbols:
+                name = expr.name
+                while name.lower() in self.used_name:
+                    name += '_'
+                self.used_name.append(name.lower())
+                if name == expr.name:
+                    self.mangled_symbols[expr] = expr
+                else:
+                    self.mangled_symbols[expr] = symbols(name, cls=expr.__class__, **expr.assumptions0)
```

Comment:
Why not use `Symbol`?

Reply:
`Symbol` was unable to make `name` an `Idx` instance.

Reply:
I think after shifting the code from `fcode` to inside of `_print_Symbol`,  the need for `cls=expr.__class__`  and `**expr.assumptions0` has vanished (i don't know why) and `Symbol` is working too now!